aviatrix/data_source_aviatrix_spoke_gateway_inspection_subnets.go

```diff
@@ -0,0 +1,59 @@
+package aviatrix
+
+import (
+	"fmt"
+
+	"github.com/AviatrixSystems/terraform-provider-aviatrix/v2/goaviatrix"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func dataSourceAviatrixSpokeGatewayInspectionSubnets() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceAviatrixSpokeGatewayInspectionSubnetsRead,
+
+		Schema: map[string]*schema.Schema{
+			"gw_name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Spoke gateway name.",
+			},
+			"subnets_for_inspection": {
+				Type:        schema.TypeList,
+				Computed:    true,
+				Description: "List of all subnets available for the subnet inspection feature.",
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+		},
+	}
+}
+
+func dataSourceAviatrixSpokeGatewayInspectionSubnetsRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*goaviatrix.Client)
+
+	gateway := &goaviatrix.Gateway{
+		GwName: d.Get("gw_name").(string),
+	}
+
+	gw, err := client.GetGateway(gateway)
```

Comment:
Why do we need to call `client.GetGateway` here if the user already passes in the `gw_name` as an attribute? Can't we just call `client.GetSubnetsForInspection` with the attribute passed in?

Reply:
Because we need to check if this spoke is in Azure or not. If we don't check, the API call later will error out.